src/components/views/rooms/UserIdentityWarning.tsx

```diff
@@ -0,0 +1,302 @@
+            if (userId === cli.getUserId()) {
+                // We always skip our own user, because we can't pin our own identity.
+                return;
+            }
+            const member = room.getMember(userId);
+            if (member) {
+                membersNeedingApprovalRef.current.set(userId, member);
+                if (!currentPrompt) {
+                    // If we're not currently displaying a prompt, then we should
+                    // display a prompt for this user.
+                    selectCurrentPrompt();
+                }
+            }
+        },
+        [cli, room, membersNeedingApprovalRef, currentPrompt, selectCurrentPrompt],
+    );
+
+    // Add a user from the membersNeedingApproval map, and update the current
+    // prompt if necessary.
+    const removeMemberNeedingApproval = useCallback(
+        (userId: string): void => {
+            membersNeedingApprovalRef.current.delete(userId);
+
+            // If we removed the currently displayed user, we need to pick a new one
+            // to display.
+            if (currentPrompt?.userId === userId) {
+                selectCurrentPrompt();
+            }
+        },
+        [membersNeedingApprovalRef, currentPrompt, selectCurrentPrompt],
+    );
+
+    // Initialise the component.  Get the room members, check which ones need
+    // their identity approved, and pick one to display.
+    const loadMembers = useCallback(async (): Promise<void> => {
+        if (!crypto || initialisedRef.current) {
+            return;
+        }
+        // If encryption is not enabled in the room, we don't need to do
+        // anything.  If encryption gets enabled later, we will retry, via
+        // onRoomStateEvent.
+        if (!(await crypto.isEncryptionEnabledInRoom(room.roomId))) {
+            return;
+        }
+        initialisedRef.current = true;
+
+        const gotVerificationStatusUpdate = gotVerificationStatusUpdateRef.current;
+        const membersNeedingApproval = membersNeedingApprovalRef.current;
+
+        const members = await room.getEncryptionTargetMembers();
+
+        for (const member of members) {
+            const userId = member.userId;
+            if (gotVerificationStatusUpdate.has(userId)) {
+                // We're already checking their verification status, so we don't
+                // need to do anything here.
+                continue;
+            }
+            gotVerificationStatusUpdate.set(userId, false);
+            if (await userNeedsApproval(crypto, userId)) {
+                if (!membersNeedingApproval.has(userId) && gotVerificationStatusUpdate.get(userId) === false) {
+                    membersNeedingApproval.set(userId, member);
+                }
+            }
+            gotVerificationStatusUpdate.delete(userId);
+        }
+
+        selectCurrentPrompt();
+    }, [crypto, room, initialisedRef, gotVerificationStatusUpdateRef, membersNeedingApprovalRef, selectCurrentPrompt]);
```

Comment:
the refs in this dependency array don't do anything: https://www.epicreact.dev/why-you-shouldnt-put-refs-in-a-dependency-array